alf/utils/common.py

```diff
@@ -433,6 +433,31 @@ def _cast(obs):
     return tf.nest.map_structure(_cast, observation)
 
 
+def traverse_transform_observation(obs, levels, field, func):
+    """
```

Comment:
Explain what this function does.

Reply:
More detailed comment: Transform the child observation in `observation` indicated by `field` using `func`